### PR TITLE
return_self_not_must_use document `#[must_use]` on the type

### PR DESCRIPTION
**Pull request by xFrednet**
Inspired by a discussion in rust-lang/rust-clippy#8197

---

r? @llogiq 

changelog: none

The lint is this on nightly, therefore no changelog entry for you xD

### CODE REVIEW

File: clippy_lints/src/return_self_not_must_use.rs [hunk 1]
```diff
@@ -65,11 +87,13 @@ fn check_method(cx: &LateContext<'tcx>, decl: &'tcx FnDecl<'tcx>, fn_def: LocalD
...
 
         then {
-            span_lint(
+            span_lint_and_help(
                 cx,
                 RETURN_SELF_NOT_MUST_USE,
                 span,
                 "missing `#[must_use]` attribute on a method returning `Self`",
+                None,
+                "consider adding the `#[must_use]` attribute to the method or `Self` type"
```
Comment (GuillaumeGomez) [inline clippy_lints/src/return_self_not_must_use.rs:96]: I would word it as is:

```suggestion
                "consider adding the `#[must_use]` attribute to the method or directly to the `Self` type"
```

File: clippy_lints/src/return_self_not_must_use.rs [hunk 2]
```diff
@@ -13,25 +14,46 @@ declare_clippy_lint! {
     /// This lint warns when a method returning `Self` doesn't have the `#[must_use]` attribute.
     ///
     /// ### Why is this bad?
-    /// It prevents to "forget" to use the newly created value.
+    /// Methods returning `Self` often create new values, having the `#[must_use]` attribute prevents
+    /// users from "forgetting" to use the newly created value.
+    ///
+    /// The `#[must_use]` attribute can be added to the struct or enum itself to ensure that instances
```
Comment (llogiq) [inline clippy_lints/src/return_self_not_must_use.rs:20]: I would write type instead of struct or enum. It's shorter and also includes unions.

**Review by llogiq — COMMENTED** [review]
Small documentation nit. I believe @GuillaumeGomez also had some suggestion on the issue.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by xFrednet** [comment]
Good suggestions, it seems like I'm not fully awake yet^^. I squashed and pushed both!

**Comment by GuillaumeGomez** [comment]
Thanks for the improvements!

**Comment by llogiq** [comment]
Thank you! @bors r+ squash

**Comment by camsteffen** [comment]
bors squash-merged it

**Pull request closed without merging**